### PR TITLE
Add division gate

### PR DESCRIPTION
**Pull request by jegork**
Hello! 

This PR adds broadcasted division operation (`/.`) for the Variable type. I am quite new to the codebase so any suggestions are welcome. 

Thanks!

### CODE REVIEW

**Comment by Vindaar** [comment]
Sorry this didn't receive any attention since January! I've unfortunately also been too busy and kept away from Github. 

Thanks for the contribution! It generally looks good to me, aside from that minor comment.

**Review by Vindaar — APPROVED** [review]
Looks good to me, just want to make sure you are certain the behavior is as expected. :) Marking it as approved anyway already!

**Comment by Clonkk** [comment]
Since it's approved should this be merged ?

**Comment by jegork** [comment]
@Clonkk sorry, completely forgot about this PR, I can take a look at the suggestions of @Vindaar or we can have it merged, what's your suggestion?

**Comment by Clonkk** [comment]
I leave it up to @Vindaar since he reviewed it. But if both of you are satisfied then it should be merged

File: src/arraymancer/autograd/gates_basic.nim [hunk 1]
```diff
@@ -96,3 +96,45 @@ proc `-`*[TT](a, b: Variable[TT]): Variable[TT] =
...
+type DivGate*[TT] {.final.} = ref object of Gate[TT]
+  a: Variable[TT]
+  b: Variable[TT]
+
+proc div_backward_ag[TT](self: Gate[TT], payload: Payload[TT]): SmallDiffs[TT] =
+  let self = DivGate[TT](self)
+  let gradient = payload.variable.grad
+  result = newDiffs[TT](2)
+  result[0] = gradient /. self.b.value
+  result[1] = - gradient *. self.a.value /. self.b.value ^. 2
```
[the remark this answers is not in the document]
Comment (Vindaar) [reply src/arraymancer/autograd/gates_basic.nim:109]: Just for future reference, here's the output of `dumpTree` for the expression:

```nim
  Command
    Ident "echo"
    Infix
      Ident "/."
      Infix
        Ident "*."
        Ident "gradient"
        Ident "a"
      Infix
        Ident "^."
        Ident "b"
        IntLit 2
  Command
    Ident "echo"
    Infix
      Ident "/"
      Infix
        Ident "*"
        Ident "gradient"
        Ident "a"
      Infix
        Ident "^"
        Ident "b"
        IntLit 2
```

meaning my fears were unfounded here. Unfortunately I don't remember which case it was that behaved unexpectedly.

**Comment by Vindaar** [comment]
Thanks and sorry again for ignoring it for such a long time :)

**Pull request merged**